zebra-state/src/service/finalized_state.rs

```diff
@@ -408,6 +491,47 @@ impl FinalizedState {
             })
     }
 
+    /// Returns the Sprout note commitment tree for a given `block::Height`
+    /// if it is present.
+    pub fn sprout_note_commitment_tree(&self) -> sprout::tree::NoteCommitmentTree {
+        let height = match self.finalized_tip_height() {
+            Some(h) => h,
+            None => return Default::default(),
+        };
+        let sprout_note_commitment_tree = self.db.cf_handle("sprout_note_commitment_tree").unwrap();
+        self.db
+            .zs_get(sprout_note_commitment_tree, &height)
+            .unwrap_or_else(Default::default)
+    }
+
+    /// Returns the Sapling note commitment tree for a given `block::Height`
+    /// if it is present.
+    pub fn sapling_note_commitment_tree(&self) -> sapling::tree::NoteCommitmentTree {
+        let height = match self.finalized_tip_height() {
+            Some(h) => h,
+            None => return Default::default(),
+        };
+        let sapling_note_commitment_tree =
+            self.db.cf_handle("sapling_note_commitment_tree").unwrap();
+        self.db
+            .zs_get(sapling_note_commitment_tree, &height)
+            .unwrap_or_else(Default::default)
```

Comment:
This seems like it could hide bugs where we accidentally reset the tree. Should we `expect` here instead?